flow/project.py

```diff
@@ -398,13 +377,216 @@ def complete(self, job):
+            def cmd(operations, directives, job=None):
+                executable = directives.get('executable', sys.executable)
+
+                if callable(executable):
+                    if job is None:
+                        raise ValueError(
+                            "Executable cannot be callable if job is not set.")
+                    else:
+                        executable = executable(job)
+
+                cmd = "{} {} run -o {}".format(executable, self.path, name)
+                if job is not None:
+                    cmd += ' -j {}'.format(job)
+                return cmd.lstrip()
+
+            self._run_cmd = cmd
+        else:
+            self._run_cmd = run_cmd
+
+        if exec_cmd is None:
+            def cmd(operations, directives, job=None):
+                if len(self.operations) > 1:
+                    # Cannot use exec mode with more than one operation.
+                    raise self.ExecCommandError
+                else:
+                    op = list(operations.values())[0]
+
+                if isinstance(op, FlowCmdOperation):
+                    return op(job).lstrip()
+                else:
+                    try:
+                        executable = directives['executable']
+                        if callable(executable):
+                            executable = executable(job)
+                    except (KeyError, TypeError):
+                        executable = sys.executable
+                    return '{} {} exec {} {}'.format(executable, self.path, op.name, job).lstrip()
+
+            self._exec_cmd = cmd
+        else:
+            self._exec_cmd = exec_cmd
+
+    def __call__(self, job=None, mode='run'):
+        "Return the string forming the command for the execution of this group."
+        if mode == 'run':
+            if callable(self._run_cmd):
+                return self._run_cmd(self.operations,
+                                     self.directives,
+                                     job) + ' ' + self.options
+            else:
+                return self._run_cmd.format(self.operations, job) + ' ' + self.options
+        elif mode == 'exec':
+            if callable(self._exec_cmd):
+                return self._exec_cmd(self.operations, self.directives, job)
+            else:
+                return self._exec_cmd.format(self.operations, job)
+        else:
+            raise ValueError("Unknown mode '{}'.".format(mode))
+
+    def __iter__(self):
+        for op in self.operations.values():
+            yield op
+
+    def eligible(self, job):
+        """Eligible, when at least one FlowOperation is eligible."""
+        return any(op.eligible(job) for op in self)
+
+    def complete(self, job):
+        "True when all FlowOperation post-conditions are met"
+        return all(op.complete(job) for op in self)
+
+    def add_operation(self, name, operation):
+        self.operations[name] = operation
+
+    def compatible(self, group):
```

Comment:
I think the word `compatible` here is not clear out of its use context and possibly even misleading. I propose we rename this to `intersects` and then use the negated form.

Reply:
Are we still using this method at all?

Reply:
Yes we are in the function `_verify_group_compatibility` on line 2033. I will rename it intersection and use the negated form though.